File: examples/python/testing-patterns/complex-workflows/test-single-child-context.py
@pytest.mark.durable_execution(handler=handler, lambda_function_name="child_context_workflow")
def test_child_context(durable_runner):
    """Test child context execution."""
    with durable_runner:
        result = durable_runner.run(input={"item_id": "item-123"}, timeout=30)
    
    assert result.status is InvocationStatus.SUCCEEDED
    
    # Check child context ran
    context_ops = [op for op in result.operations if op.operation_type.value == "CONTEXT"]
    assert len(context_ops) == 1
    assert context_ops[0].name == "item_processing"
    
    # Check child context result
    child_result = result.get_context("item_processing")
    child_data = deserialize_operation_payload(child_result.result)
    assert child_data["item_id"] == "item-123"

File: examples/java/testing-patterns/complex-workflows/test-single-child-context.java
@Test
void testChildContext() {
    var runner = LocalDurableTestRunner.create(Map.class,
        new SingleChildContextHandler());

    var result = runner.runUntilComplete(Map.of("item_id", "item-123"));

    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

    // Check child context ran
    var contextOp = result.getOperation("item_processing");
    assertNotNull(contextOp);
    assertEquals(OperationStatus.SUCCEEDED, contextOp.getStatus());

    // Check child context result
    var childResult = contextOp.getStepResult(Map.class);
    assertEquals("item-123", childResult.get("item_id"));
}
